diff --git a/src/writer.js b/src/writer.js
index 7b04f78..1da761e 100644
--- a/src/writer.js
+++ b/src/writer.js
@@ -75,6 +75,16 @@ async function writeMarkdownFilesPromise(posts, config ) {
 async function loadMarkdownFilePromise(post) {
 	let output = '---\n';
 
+	if(post.frontmatter.categories) {
+		post.frontmatter.tags = [...post.frontmatter.tags , ...post.frontmatter.categories];
+		delete post.frontmatter.categories;
+	}
+	if(post.frontmatter.tags) {
+		post.frontmatter.tags.push("post");
+	} else {
+		post.frontmatter.tags = ["post"];
+	}
+
 	Object.entries(post.frontmatter).forEach(([key, value]) => {
 		let outputValue;
 		if (Array.isArray(value)) {
@@ -82,10 +92,12 @@ async function loadMarkdownFilePromise(post) {
 				// array of one or more strings
 				outputValue = value.reduce((list, item) => `${list}\n  - "${item}"`, '');
 			}
-		} else {
+		} else if(key != 'date') {
 			// single string value
 			const escapedValue = (value || '').replace(/"/g, '\\"');
 			outputValue = `"${escapedValue}"`;
+		} else {
+			outputValue = value;
 		}
 
 		if (outputValue !== undefined) {
@@ -93,6 +105,8 @@ async function loadMarkdownFilePromise(post) {
 		}
 	});
 
+	output += `slug: ${post.meta.slug}\n`;
+
 	output += `---\n\n${post.content}\n`;
 	return output;
 }
